src/applications/vaos/components/PastAppointmentsList.jsx

```diff
@@ -1,91 +1,158 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
 import LoadingIndicator from '@department-of-veterans-affairs/formation-react/LoadingIndicator';
 import AlertBox from '@department-of-veterans-affairs/formation-react/AlertBox';
+import { fetchPastAppointments } from '../actions/appointments';
 import { FETCH_STATUS, APPOINTMENT_TYPES } from '../utils/constants';
-import { getAppointmentType } from '../utils/appointment';
+import { vaosPastAppts } from '../utils/selectors';
+import {
+  getAppointmentType,
+  getPastAppointmentDateRangeOptions,
+} from '../utils/appointment';
 import ConfirmedAppointmentListItem from './ConfirmedAppointmentListItem';
 import PastAppointmentsDateDropdown from './PastAppointmentsDateDropdown';
+import TabNav from './TabNav';
 
-export default function PastAppointmentsList({
-  appointments,
-  onDateRangeChange,
-  dateRangeOptions,
-  selectedDateRangeIndex,
-}) {
-  const { past, pastStatus, systemClinicToFacilityMap } = appointments;
-  const loading = pastStatus === FETCH_STATUS.loading;
-  const hasAppointments =
-    pastStatus === FETCH_STATUS.succeeded && past?.length > 0;
-  let content;
+const dateRangeOptions = getPastAppointmentDateRangeOptions();
 
-  if (loading) {
-    content = (
-      <div className="vads-u-margin-y--8">
-        <LoadingIndicator message="Loading your appointments..." />
-      </div>
+export class PastAppointmentsList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedDateRangeIndex: this.props.appointments.pastSelectedIndex || 0,
+      selectedDateRange: dateRangeOptions[0],
```

Comment:
Should this be `this.props.appointments.pastSelectedIndex` instead of 0? Also, do you need this in state if you have the index already?

Also, does this need to be in component state at all? It's always a little tricky to sync up Redux state and local component state.

Reply:
Yeah, I wasn't sure about this.  The intent of storing the index in redux was to prevent a repeat fetch if the user was to switch back and forth between tabs.   I could potentially keep track of this in AppointmentsPage or just disregard it completely. What do you think?

Reply:
I think storing the index in Redux is fine, but it doesn't seem like you need it in the component state?